src/indexer/merger.rs

```diff
@@ -927,19 +928,25 @@ impl IndexMerger {
             .collect();
         if let Some(doc_id_mapping) = doc_id_mapping {
             for (old_doc_id, reader_with_ordinal) in doc_id_mapping {
-                let store_reader = &mut document_iterators[reader_with_ordinal.ordinal as usize];
-                let raw_doc = store_reader.next().expect(&format!(
-                    "unexpected missing document in docstore on merge, doc id {:?}",
-                    old_doc_id
-                ))?;
-                store_writer.store_bytes(raw_doc.get_bytes())?;
+                let doc_bytes_it = &mut document_iterators[reader_with_ordinal.ordinal as usize];
+                if let Some(doc_bytes_res) = doc_bytes_it.next() {
+                    let doc_bytes = doc_bytes_res?;
+                    store_writer.store_bytes(&doc_bytes)?;
+                } else {
+                    return Err(DataCorruption::comment_only(&format!(
+                        "unexpected missing document in docstore on merge, doc id {:?}",
+                        old_doc_id
+                    ))
+                    .into());
```

Comment:
ok_or_else would be nicer here

https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or_else

As a general note, I think it would be good to add backtraces to the Errors. If something fails this can be really helpful. It seems like `thiserror` supports this

Reply:
I used ok_or_else at first and judged that the double ?? was confusing

Reply:
backtrace would be useful indeed.

Reply:
since this reads like a book, more `?` are better for dramatic purposes

`let it_was_not_an_error = is_it_an_error???;`

Reply:
:)